addons/payment/models/payment_transaction.py

```diff
@@ -131,6 +141,16 @@ def _compute_invoices_count(self):
         for tx in self:
             tx.invoices_count = tx_data.get(tx.id, 0)
 
+    def _compute_refunds_count(self):
+        rg_data = self.env['payment.transaction'].read_group(
+            domain=[('source_transaction_id', 'in', self.ids), ('operation', '=', 'refund')],
+            fields=['source_transaction_id'],
+            groupby=['source_transaction_id'],
+        )
```

Comment:
About `source_transaction_id`, since it's only ever used in a stat button for 2 views (payment and transaction, which should rarely be open), we think that the cost overweight the gain.
But I'll look for `state` and `operation`, they are used outside the refund flow, so, I think they'll benefit from an index.